src/utils/SoAStorage.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file SoAStorage.h
+ * @author seckler
+ * @date 11.06.18
+ */
+
+#pragma once
+
+#include <tuple>
+
+namespace autopas {
+namespace utils {
+
+/**
+ * SoAStorage is a helper to access the stored SoA's.
+ * @tparam SoAArraysType the type of the storage arrays. should be a tuple of aligned vectors
+ */
+template <class SoAArraysType>
+class SoAStorage {
+ private:
+  // End of iteration.
+  template <std::size_t I = 0, typename FuncT, typename... Tp>
+  inline typename std::enable_if<I == sizeof...(Tp), void>::type for_each(std::tuple<Tp...>&, FuncT) {}
```

Comment:
Out of curiosity: Does an `inline`d recursive function have the same performance as an iterative?

Reply:
As it is a template meta function, there is no performance penalty at all. I even doubt that it calls any functions. It probably just inlines everything.
This is definitely faster than the previous iteration of the vectors inside the map.
But it anyways is not performance relevant, as it just iterates over the tuple of vectors and not within the vectors, anyways.